partner_event/models/event_registration.py

```diff
@@ -34,16 +35,19 @@ def create(self, vals):
             Event = self.env['event.event']
             # Look for a partner with that email
             email = vals.get('email').replace('%', '').replace('_', '\\_')
+            event = Event.browse(vals['event_id'])
+            company_id = event.company_id and event.company_id.id
             attendee_partner = Partner.search([
-                ('email', '=ilike', email)
+                ('email', '=ilike', email),
+                ('company_id', '=', company_id),
```

Comment:
```suggestion
                '|',
                ('company_id', '=', company_id),
                ('company_id', '=', False),
```

Reply:
Other better solution can be a second search with ('company_id', '=', False) after first search if it don't find results

Reply:
Well, indeed if the company_id is False, that means that the partner is shared across companies, so it should be fine, isn't it?

Reply:
Yes, but it's for the improbable case of having 2 equal partners, one with the company set, and another without, giving priority to the specific company partner, but I think it's to fine grain a lot. What do you think?